assets/src/data/model/entity-factory/constants.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External imports
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { isUndefined } from 'lodash';
+
+/**
+ * Constants describing the current "save state" for an entity.
+ *
+ * @type {{CLEAN: Symbol, NEW: Symbol, DIRTY: Symbol}}
+ */
+export const SAVE_STATE = {
+	CLEAN: Symbol( 'Entity is persisted.' ),
+	NEW: Symbol( 'Entity is new.' ),
+	DIRTY: Symbol( 'Existing entity has changes and needs persisted.' ),
+};
+
+/**
+ * Validation types are for schema's that have value variations.
+ * @type {{RAW: string, RENDERED: string, PRETTY: string}}
+ */
+export const VALIDATE_TYPE = {
+	RAW: 'raw',
+	RENDERED: 'rendered',
+	PRETTY: 'pretty',
+};
+
+/**
+ * Private properties used internally by the Base Entity Class
+ * @type {{saveState: boolean}}
+ */
+export const PRIVATE_PROPERTIES = {
+	SAVE_STATE: Symbol( 'baseEntityPrivatePropertiesSaveState' ),
+	VALIDATE_TYPES: Symbol( 'baseEntityPrivatePropertiesValidateTypes' ),
+};
+
+/**
+ * Hardcoded list of model prefixes for fields on models.
+
+ * A model prefix is something that "namespaces" a field on a model.  For
+ * example, if the field is "EVT_ID", then the prefix is "EVT"; if the field is
+ * "DTT_EVT_start", then the prefixes are "DTT", and "DTT_EVT".
+ *
+ * @param  {string} modelName
+ * @return {Object} A filtered object indexed by model name and the values are
+ * an array of model prefixes for that model.
+ */
+export const MODEL_PREFIXES = ( modelName ) => {
+	const prefixMap = applyFilters(
```

Comment:
unfortunately there is no exposure of model entity prefixes on our REST API or elsewhere.  For now this is hardcoded (with a filter for add-ons to use) but it may be something we'll want to consider exposing on the REST API somehow.